reVX/plexos/plexos_plants.py

```diff
@@ -0,0 +1,1113 @@
+            Location (lat, lon), final capacity, and associated sc_gids,
+            res_gids, and res gid_counts for all plants
+        """
+        plants_meta = []
+        for pid, plant in self.plants.items():
+            plants_meta.append(pd.Series(
+                {'sc_gids': plant['sc_gid'].values.tolist(),
+                 'res_gids': plant['res_gids'].values.tolist(),
+                 'gid_counts': plant['gid_counts'].values.tolist(),
+                 'res_cf_means': plant['cf_means'].values.tolist(),
+                 'build_capacity': plant['build_capacity'].values.tolist()},
+                name=pid))
+
+        plants_meta = pd.concat(plants_meta, axis=1).T
+        plants_meta.index.name = 'plant_id'
+
+        if out_fpath:
+            plants_meta.to_csv(out_fpath)
+
+        return plants_meta
+
+    @classmethod
+    def save(cls, plexos_table, sc_table, cf_fpath, out_fpath,
+             dist_percentile=90, lcoe_col='total_lcoe', lcoe_thresh=1.3,
+             offshore=False, max_workers=None, plants_per_worker=40,
+             points_per_worker=400):
+        """
+        Identify, fill, and then save plants to disc
+
+        Parameters
+        ----------
+        plexos_table : pandas.DataFrame
+            Parsed and clean PLEXOS table
+        sc_table : str | pandas.DataFrame
+            Supply Curve table .csv or pre-loaded pandas DataFrame
+        cf_fpath : str
+            Path to reV generation output .h5 file
+        out_fpath : str
+            .csv path to save plant meta data too
+        offshore : bool, optional
+            Include offshore points, by default False
+        dist_percentile : int, optional
+            Percentile to use to compute distance threshold using sc_gid to
+            SubStation distance , by default 90
+        lcoe_col : str, optional
+            LCOE column to sort by, by default 'total_lcoe'
+        lcoe_thresh : float, optional
+            LCOE threshold multiplier, exclude sc_gids above threshold,
+            by default 1.3
+        offshore : bool, optional
+            Include offshore points, by default False
+        max_workers : int, optional
+            Number of workers to use for plant identification, 1 == serial,
+            > 1 == parallel, None == parallel using all available cpus,
+            by default None
+        plants_per_worker : int, optional
+            Number of plants to identify on each worker, by default 40
+        points_per_worker : int, optional
+            Number of points to create on each worker, by default 400
+        """
+        pp = cls(plexos_table, sc_table, cf_fpath,
+                 dist_percentile=dist_percentile, lcoe_col=lcoe_col,
+                 lcoe_thresh=lcoe_thresh, offshore=offshore,
+                 max_workers=max_workers, plants_per_worker=plants_per_worker,
+                 points_per_worker=points_per_worker)
+
+        pp.dump(out_fpath=out_fpath)
+
+
+class PlantProfileAggregation:
```

Comment:
I think this division of classes makes a lot more sense. I still think you should have used the pre-existing PlexosNode object but thats okay.

Reply:
This was more difficult than it seems. The only bit of code that is really shared with PlexosNode is in this method:
https://github.com/NREL/reVX/blob/423e0abf7614f73639ebaa4e57ce1a30b82c377f/reVX/plexos/plexos_plants.py#L962
But that is embedded within the PlexosNode aggregation logic here: 
https://github.com/NREL/reVX/blob/423e0abf7614f73639ebaa4e57ce1a30b82c377f/reVX/plexos/rev_reeds_plexos.py#L162

The rest is unique to the Plexos Plant input.

I love the idea of generalizing some of this, but its tough with how unique (and different) the inputs are from ReEDS vs Plexos.  Down the road, we could try to generalize this profile making bit and maybe the output code to take any set of datasets.